pkg/staticnetworkconfig/generator.go

```diff
@@ -167,7 +168,8 @@ func (s *StaticNetworkConfigGenerator) formatNMConnection(nmConnection string) (
 
 func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistence(macInterfaceMap models.MacInterfaceMap, networksConfig []StaticNetworkConfigData) error {
 	interfaceNames := lo.Map(macInterfaceMap, func(m *models.MacInterfaceMapItems0, _ int) string { return m.LogicalNicName })
-	var matched bool
+	// See 'man systemd.net-naming-scheme' for interface naming protocol
+	predicatableIfaceNamePattern := regexp.MustCompile("^en[PsvxXbucaipod]")
```

Comment:
I think this will catch most physical interface names, we should note however that some invalid physical interface names will be matched by this, some examples...

`enxABC` - The `enx` prefix is used to denote a physical interface with a MAC address, MAC addresses should have exactly 12 characters.

`enp0` - PCI bus specified but no device specified

and other similar combinations where we include any of the parameter names used for physical interfaces.

Should we make the regex more specific?

Reply:
I believe the intent was to detect names that would be used as placeholders and require the interface-mapping in that case, not to validate all potential physical interface names as discussed in https://issues.redhat.com/browse/OCPBUGS-34849?focusedId=25452379&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-25452379.

Reply:
setting `biosdevname=1` will also can break thios match